packages/core/src/model/nodes/MCPNode.ts

```diff
@@ -0,0 +1,603 @@
+
+    const parts = [base];
+
+    if (this.data.communicationMode === 'stdio' && context.executor !== 'nodejs') {
+      parts.push('(Requires Node Executor)');
+    }
+
+    if (this.data.availableTools?.length) {
+      parts.push(`(${this.data.availableTools.length} tools)`);
+    }
+
+    return parts.join(' ');
+  }
+
+  async process(inputs: Inputs, context: InternalProcessContext): Promise<Outputs> {
+    try {
+        const input = coerceType(inputs['input' as PortId], 'object');
+        const serverId = this.data.communicationMode === 'stdio' 
+            ? coerceType(inputs['serverId' as PortId], 'string') 
+            : this.data.serverId;
+        const config = this.data.communicationMode === 'stdio' 
+            ? inputs['config' as PortId] 
+            : this.data.configuration;
+
+        // Process the inputs as needed
+        const response = await this.handleCommunication(inputs, input, context);
+        return this.formatSuccessResponse(response);
+    } catch (error) {
+        return this.formatErrorResponse(error as MCPError);
+    }
+  }
+
+  private async handleCommunication(
+    inputs: Inputs,
+    input: object,
+    context: InternalProcessContext
+  ): Promise<{ output: string; metadata: Record<string, unknown> }> {
+    if (this.data.communicationMode === 'http') {
+      return this.handleHttpCommunication(inputs, input);
+    }
+    return this.handleStdioCommunication(inputs, input, context);
+  }
+
+  private async handleHttpCommunication(inputs: Inputs, input: object): Promise<{ output: string; metadata: Record<string, unknown> }> {
+    const endpoint = this.data.useEndpointInput
+      ? coerceType(inputs['endpoint' as PortId], 'string')
+      : this.data.endpoint;
+
+    const headers = this.data.useHeadersInput
+      ? (inputs['headers' as PortId] as { type: 'object'; value: Record<string, string> })?.value ?? {}
+      : Object.fromEntries(this.data.headers.map(({ key, value }) => [key, value]));
+
+    const response = await fetch(endpoint, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        ...headers,
+      },
+      body: JSON.stringify({
+        input,
+        configuration: this.data.configuration,
+      }),
+    });
+
+    if (!response.ok) {
+      throw new MCPError(MCPErrorType.HTTP_ERROR, `HTTP error! status: ${response.status}`);
+    }
+
+    return response.json();
+  }
+
+  private async handleStdioCommunication(
+    inputs: Inputs,
+    input: object,
+    context: InternalProcessContext
+  ): Promise<{ output: string; metadata: Record<string, unknown> }> {
+    let configFile: object;
+    try {
+      configFile = this.data.useEndpointInput
+        ? JSON.parse(coerceType(inputs['config' as PortId], 'string'))
+        : {};
+    } catch (error) {
+      throw new MCPError(MCPErrorType.INVALID_RESPONSE, 'Invalid format for MCP configuration');
+    }
+
+    const serverId = this.data.useEndpointInput
```

Comment:
You can use `getInputOrData` for this pattern, `getInputOrData(this.data, inputs, 'serverId')`